src/pages/home/report/ReportActionsView.js

```diff
@@ -68,15 +71,16 @@ class ReportActionsView extends React.Component {
         this.renderItem = this.renderItem.bind(this);
         this.renderCell = this.renderCell.bind(this);
         this.scrollToListBottom = this.scrollToListBottom.bind(this);
-        this.recordMaxAction = this.recordMaxAction.bind(this);
         this.onVisibilityChange = this.onVisibilityChange.bind(this);
         this.loadMoreChats = this.loadMoreChats.bind(this);
         this.sortedReportActions = [];
         this.timers = [];
-
-        this.initialNewMarkerPosition = props.report.unreadActionCount === 0
+        const newMarkerPosition = props.report.unreadActionCount === 0
             ? 0
-            : (props.report.maxSequenceNumber + 1) - props.report.unreadActionCount;
+            : props.report.maxSequenceNumber - props.report.unreadActionCount;
+
+        // Set the new marker
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${props.reportID}`, {newMarkerPosition});
```

Comment:
Per the [Expensify.cash Philosophy](https://github.com/Expensify/Expensify.cash#philosophy), only actions can manage Onyx data, not components.

Reply:
So I think instead you can just do:

```javascript
setLocalLastRead(props.reportID, newMarkerPosition, true)
```

right?